sync/sync_test.go

```diff
@@ -349,6 +349,36 @@ func TestSync_ShouldRegisterMissingRegistrationInMultiregistrationScenario(t *te
 	assert.Len(t, services, 2)
 }
 
+/*
+This may happen if an application configuration is changed and then the deployment this triggered is rolled back.
+After rollback, the new configuration still stays with the application and there's no way to access the original configuration
```

Comment:
Actually this may happen in many ways. E.g. you change configuration, deployment starts, but for some reason the application cannot start so deployment spins endlessly. Configuration gets updated though, so during sync the existing tasks refer to this new configuration. That's even before you click deployment stop. I'm going to update this documentation comment to reflect this.